src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -900,3 +900,12 @@ void QgsRelationReferenceWidget::disableChainedComboBoxes( const QComboBox *scb
     ccb = cb;
   }
 }
+
+void QgsRelationReferenceWidget::emitForeignKeyChanged( const QVariant &foreignKey )
+{
+  if ( foreignKey != mForeignKey || foreignKey.isNull() != mForeignKey.isNull() )
```

Comment:
Under what circumstances the second condition may be true if the first one is false?

Reply:
Unfortunately `QVariant( 0 ) == QVariant ( QVariant::Int ) /*Null*/`